router/client/src/sharded_client.rs

```diff
@@ -174,8 +184,37 @@ impl ShardedClient {
             .iter_mut()
             .map(|client| Box::pin(client.classify(batch.clone())))
             .collect();
-        let results: Result<Vec<Vec<EntityList>>> = join_all(futures).await.into_iter().collect();
-        Ok(results?.into_iter().flatten().collect())
+        let results: Result<Vec<Vec<ClassifyPredictionList>>> =
+            join_all(futures).await.into_iter().collect();
+
+        let flat_results: Vec<ClassifyPredictionList> = results?.into_iter().flatten().collect();
+        let threaded_futures: Vec<_> = flat_results
```

Comment:
I would be very curious to time this block of code to see how long it takes. My concern with this approach is that even though we're parallelizing the `format_ner_outputs` operation, we're doing so in a way that is blocking the batching thread, so during this period of time, the GPU is entirely idle.

Instead, we should move this logic into the request thread here:

https://github.com/predibase/lorax/blob/main/router/src/infer.rs#L659-L661

In that way, we do not block the batching thread so we can move on to the next batch and keep the GPU busy.

Does that make sense? Happy to quickly sync on this if needed.

Reply:
addressed